kafka-plugins-common/src/main/java/io/cdap/plugin/kafka/batch/source/KafkaPartitionOffsets.java

```diff
@@ -47,7 +47,7 @@ public KafkaPartitionOffsets(Map<Integer, Long> partitionOffsets) {
   }
 
   public void setPartitionOffset(int partition, long offset) {
-    partitionOffsets.computeIfPresent(partition, (k, v) -> offset);
+    partitionOffsets.putIfAbsent(partition, offset);
```

Comment:
Will a simple put work ? Is there a concern about overwriting?

Reply:
Yes, a simple `put()` works as well. No issues with overwriting. Changed to `put()` 981e29e3cfc2b26f6a01c547c6b33b9d548ce680

Reply:
> Please add unit tests. Is the fix also applicable for streaming source ?

The fix is only applicable for the Batch Source. This bug [PLUGIN-1594](https://cdap.atlassian.net/browse/PLUGIN-1594) was not observed in Kafka Streaming Source Plugin.

Reply:
> Please add unit tests. Is the fix also applicable for streaming source ?

Added a unit test. 23faa7897eb4469b19ff1cb7d204d6139b3c5b42